tests/kernel/device/src/main.c

```diff
@@ -248,14 +279,39 @@ void test_dummy_device_pm(void)
 	busy = device_busy_check(dev);
 	zassert_true((busy == 0), NULL);
 
+	test_build_suspend_device_list();
+
+	/* Set device state to DEVICE_PM_ACTIVE_STATE */
+	ret = device_set_power_state(dev, DEVICE_PM_ACTIVE_STATE, NULL, NULL);
+	if (ret == -ENOTSUP) {
+		zassert_true((ret == -ENOTSUP),
+			     "Power management not supported on device");
```

Comment:
Same as line 228.

Reply:
It is in order to handle the case that the device is not support power management. I make a change here to add an "else" branch to tell the different case.

Reply:
as previous comment, I modified the code logic here as pagibot comment, I think it's good to do this way.